Library/Homebrew/extend/os/mac/hardware/cpu.rb

```diff
@@ -107,6 +107,18 @@ def universal_archs
         end
       end
 
+      # Determines whether the current CPU and macOS combination
+      # can run an executable of the specified architecture.
+      # `arch` is a symbol in the same format returned by
+      # Hardware::CPU.family
+      def can_run? arch
+        if Hardware::CPU.intel?
+          intel_can_run? arch
+        else
+          ppc_can_run? arch
+        end
```

Comment:
Maybe:
```
elsif Hardware::CPU.ppc?
  ppc_can_run? arch
else
  false
end
```

The idea is to avoid unnecessary (and maybe incorrect) assumption that it is either only intel or ppc?